### ISSUE TITLE
cargo-apk compilation problem

### ISSUE THREAD

**Issue by hanxiaohuivc** [rust-mobile/cargo-apk#23]
cargo-apk compilation problem
Excuse me, when I compile using the "cargo apk build" command on both windows and linux, the following error occurs: Error: Path `"/home/develop/cargo-apk/target/aarch64-linux-android/debug/libndk_build.so" ` doesn't exist.
What is the cause of this? How to compile it correctly?
Thanks.

**Comment by JohnScience** [comment]
@hanxiaohuivc In my case, I had to rename the name of the library. `cargo apk` doesn't seem to respect `[lib]` section in `Cargo.toml`.

**Comment by MarijnS95** [comment]
> `cargo apk` doesn't seem to respect `[lib]` section in `Cargo.toml`.

It does since #26, but that hasn't been released yet because this crate is officially deprecated in favour of `xbuild`.

Perhaps I'll do one more final release with the last few contributions, and a clear marker at the top of the `README`.

---

The error @hanxiaohuivc is facing sounds like they have not set their `[lib]` to compile as a `cdylib`. We now have more robust checking/forcing for that in #28. Though strangely `ndk-build` is also the name of the support crate for `cargo-apk`, both of which are _host tools_ and should unlikely be cross-compiled for Android (unless you're running this from Termux, but then you _should not_ be using `cargo apk` because they are plain libs/exes, not an APK).

**Comment by mvvvv** [comment]
I have the same problem since I move my main.rs in examples directory.
Error: Path `"/home/mv/dvlt/abacaba/abacaba/target/aarch64-linux-android/debug/examples/libskybox1.so"` doesn't exist.

**Comment by MarijnS95** [comment]
@mvvvv have you configured your examples to compile as `cdylib` as well? See how the examples _within this repo_ do it:

https://github.com/rust-mobile/cargo-apk/blob/558adff4f77740bc3a9353e2a8350064fa2942bd/examples/Cargo.toml#L17-L19

**Comment by mvvvv** [comment]
Thanks, it works perfectly

**Comment by mvvvv** [comment]
Too bad. Now I cannot launch my example compiled for linux.
` [[example]] 
 name = "hello_world" 
 crate-type = ["bin"] `
works only for linux

` [[example]] 
 name = "hello_world" 
 crate-type = ["bin","cdylib"] `
works only for android.

Any Idea ?

**Comment by mvvvv** [comment]
Something realy ugly is that with ["bin"] if an old file aarch64 libskybox1.so was previously successfully built, then this old file is packaged in the apk and the build is successfull.

**Comment by MarijnS95** [comment]
Agreed, it's ugly and unfortunate but not something we can very easily work around. Having two examples is likely your best bet for the time being.

**Comment by mvvvv** [comment]
I've found the solution. I create a symbolic link of the example, set te example crate-type to "cdylib" and run the link for PC tests. it works well with VS code and launch.json.

**Comment by MarijnS95** [comment]
Such tricks also exist with `path=` in `Cargo.toml`. It's ugly but hard to work around until we get better integration with `cargo`.
